packages/http/src/route.ts

```diff
@@ -216,28 +216,43 @@ export function setRoute(context: DecoratorContext, entity: Type, details: Route
   const state = context.program.stateMap(routesKey);
 
   if (state.has(entity) && entity.kind === "Namespace") {
-    const existingValue: RoutePath = state.get(entity);
-    if (existingValue.path !== details.path) {
+    const existingPath: string | undefined = state.get(entity);
+    if (existingPath !== details.path) {
       reportDiagnostic(context.program, {
         code: "duplicate-route-decorator",
         messageId: "namespace",
         target: entity,
       });
     }
   } else {
-    state.set(entity, details);
+    state.set(entity, details.path);
+    if (entity.kind === "Operation" && details.shared) {
+      setSharedRoute(context.program, entity as Operation);
+    }
```

Comment:
I think we should deprecate this pattern if we are going with an `@sharedRoute` decorator

Reply:
I would also like to deprecate it if we proceed with `@sharedRoute`, but I'm not sure how to deprecate a parameter to a decorator at this moment.  A simple diagnostic could work, I suppose.

Reply:
That, and you should be able to mark it with `@deprecated` in the docstring, I think. @timotheeguerin would know for sure.

Reply:
You can just call `reportDeprecated` but with a condition on the parameter being used instead of just every time for the decorator

Reply:
Even better, we should add a migration script to do this for people

Reply:
Added deprecation for the old `shared` option pattern.